robolectric/src/main/java/org/robolectric/android/internal/AndroidTestEnvironment.java

```diff
@@ -682,16 +688,56 @@ private String createTempDir(String name) {
         .toString();
   }
 
+  private static BroadcastReceiver newBroadcastReceiverFromP(
+      String receiverClassName,
+      Application application,
+      LoadedApk loadedApk,
+      BroadcastReceiverData receiver) {
+    ClassLoader classLoader;
+    if (application.getBaseContext() != null) {
+      classLoader = application.getBaseContext().getClassLoader();
+    } else {
+      classLoader = loadedApk.getClassLoader();
+    }
+    for (String action : receiver.getActions()) {
+      try {
+        // If one BroadcastReceiver has multiple actions, we will try to load it with action one
+        // by one until we can initialize this BroadcastReceiver correctly.
+        Intent intent = new Intent(action);
```

Comment:
Why not just pass null for the intent? Is this modeled after some logic in Android? Seems like AppComponentFactory should be able to handle null intents

Reply:
Oh, sorry for that. I forget it. The method signature of `AppComponentFactory` allows null intent. I will fix it with above issues together at this weekend.

Reply:
Sounds good! I did some tests and it seems like a null intent works fine for this. Also we will not need a system property after all :)